src/component/handlers/composition/DOMObserver.js

```diff
@@ -124,7 +128,11 @@ class DOMObserver {
         return target.textContent;
       }
     } else if (type === 'childList') {
-      if (removedNodes && removedNodes.length) {
+      if (addedNodes && addedNodes.length) {
+        // This mutation is creating a new node, meaning it's happening
+        // at the beginning of a leaf, so we must force a re-render.
+        this.mutationAtLeafStart = true;
```

Comment:
nit, I'd add the `has` prefix to clarify that this is a boolean.
```suggestion
        this.hasMutationAtLeafStart = true;
```

Reply:
Hey, @fcosta-godaddy. Thank you for the feedback. Done!